audbackend/core/utils.py

```diff
@@ -78,6 +80,66 @@ def check_version(version: str) -> str:
     return version
 
 
+def checksum(file: str) -> str:
+    r"""Checksum of file.
+
+    The checksum is given by the MD5 sum
+    as calculated with :func:`audeer.md5`.
+
+    As parquet files are stored non-deterministically,
```

Comment:
I have successlessly tried to find details on how the  metadata hash field can be determined without importing ` pyarrow`.  

I think that the idea of the implementation could be a little clearer in the docstring

- the default way of calculating hashes is  `audeer.md5` 
- For parquet files there can be departures in situations in which pyarrow is importable and the metadata contain a hash field. 

So in short I wouldn't call it "allow". Rather make the conditions explicit in which the pyarrow way of hashing is pursued.

Reply:
You have written it quite nicely in the MR description: 

"The checksum for parquet is given by its "hash" metadata entry, or if not available by audeer.md5()."

Reply:
I shortened an updated the docstring description to

![image](https://github.com/user-attachments/assets/dd27f285-19a9-4910-b813-66244973cdeb)

Reply:
For me this has become much clearer now. I am resolving this thread.